edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/DockerStats.cs

```diff
@@ -4,98 +4,170 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Docker.Models
+        {
+            this.Name = Option.Maybe(name);
+            this.BlockIoStats = Option.Maybe(block_io_stats);
+            this.CpuStats = Option.Maybe(cpu_stats);
+            this.MemoryStats = Option.Maybe(memory_stats);
+            this.Networks = Option.Maybe(networks);
+            this.NumProcesses = Option.Maybe(num_processes);
+            this.PidsStats = Option.Maybe(pids_stats);
+            this.Read = Option.Maybe(read);
+
+            // Check for null entries in dictionarys
+            if (this.BlockIoStats.Exists(n => n.Any(o => o.Key == null || o.Value == null)))
+            {
+                this.BlockIoStats = Option.None<Dictionary<string, DiskIO[]>>();
+            }
+
+            if (this.Networks.Exists(n => n.Any(o => o.Key == null || o.Value == null)))
+            {
+                this.Networks = Option.None<Dictionary<string, NetworkInfo>>();
+            }
+        }
+
         [JsonProperty("name", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public string Name { get; set; }
+        public Option<string> Name { get; }
 
         [JsonProperty("blkio_stats", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public Dictionary<string, DiskIO[]> BlockIoStats { get; set; }
+        public Option<Dictionary<string, DiskIO[]>> BlockIoStats { get; }
 
         [JsonProperty("cpu_stats", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public DockerCpuStats CpuStats { get; set; }
+        public Option<DockerCpuStats> CpuStats { get; }
 
         [JsonProperty("memory_stats", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public MemoryStats MemoryStats { get; set; }
+        public Option<MemoryStats> MemoryStats { get; }
 
         [JsonProperty("networks", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public Dictionary<string, NetworkInfo> Networks { get; set; }
+        public Option<Dictionary<string, NetworkInfo>> Networks { get; }
 
         [JsonProperty("num_procs", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public int NumProcesses { get; set; }
+        public Option<int> NumProcesses { get; }
 
         [JsonProperty("pids_stats", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public PidsStats PidsStats { get; set; }
+        public Option<PidsStats> PidsStats { get; }
 
         [JsonProperty("read", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
-        public DateTime Read { get; set; }
+        public Option<DateTime> Read { get; }
     }
 
     public class DiskIO
     {
+        [JsonConstructor]
+        public DiskIO(long? major, long? minor, string op, long? value)
+        {
+            this.Major = Option.Maybe(major);
```

Comment:
If all fields are optional, but just create Option.None<DiskIO>().  Some fields should be required, right?

Reply:
Please put a comment to the class, mentioned this class is just used for serializing docker stats result.

Reply:
added